src/installer/tests/HostActivation.Tests/PortableAppActivation.cs

```diff
@@ -554,6 +554,40 @@ public void AppHost_GUI_FrameworkDependent_MissingRuntimeFramework_ErrorReported
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void AppHost_GUI_RuntimeMissing_PrintArchitectureAndRuntimeVersionOnDialog()
+        {
+            var fixture = sharedTestState.PortableAppFixture_Built
+                .Copy();
+
+            string appExe = fixture.TestProject.AppExe;
+            File.Copy(sharedTestState.BuiltAppHost, appExe, overwrite: true);
+            AppHostExtensions.BindAppHost(appExe);
+            AppHostExtensions.SetWindowsGraphicalUserInterfaceBit(appExe);
+
+            string invalidDotNet = SharedFramework.CalculateUniqueTestDirectory(Path.Combine(TestArtifact.TestArtifactsPath, "guiErrors"));
+            Directory.CreateDirectory(invalidDotNet);
+            File.Copy(sharedTestState.BuiltDotNet.DotnetExecutablePath, Path.Combine(invalidDotNet, "dotnet.exe"), true);
+
+            var command = Command.Create(appExe)
+                .EnableTracingAndCaptureOutputs()
+                .DotNetRoot(invalidDotNet)
+                .MultilevelLookup(false)
+                .Start();
+
+            WaitForPopupFromProcess(command.Process);
+            command.Process.Kill();
+
+            var expectedErrorCode = Constants.ErrorCode.CoreHostLibMissingFailure.ToString("x");
+            var result = command.WaitForExit(true)
+                .Should().Fail()
+                .And.HaveStdErrContaining($"Showing error dialog for application: '{Path.GetFileName(appExe)}' - error code: 0x{expectedErrorCode}")
+                .And.HaveStdErrContaining($"url: 'https://aka.ms/dotnet-core-applaunch?missing_runtime=true")
+                .And.HaveStdErrContaining("gui=true")
+                .And.HaveStdErrContaining($"&apphost_version={sharedTestState.RepoDirectories.MicrosoftNETCoreAppVersion}");
```

Comment:
This test is largely overlapping with AppHost_GUI_FrameworkDependent_MissingRuntimeFramework_ErrorReportedInDialog above where missingHostfxr=true. I think it would make sense to just remove the missingHostfxr=true case from the test above so that it is just the test for a missing framework (NETCoreApp) and this test can be the one for missing runtime (hostfxr).

Reply:
Split above's test into AppHost_GUI_FrameworkDependent_MissingRuntimeFramework_ErrorReportedInDialog and AppHost_GUI_MissingRuntime_ErrorReportedInDialog. The two are pretty much the same only difference are the URL that should be produced and the error code -- I think having the two would be more valuable if we had a way to check the dialog text.